lib/kredis/types/ordered_set.rb

```diff
@@ -0,0 +1,71 @@
+class Kredis::Types::OrderedSet < Kredis::Types::Proxying
+  proxying :multi, :zrange, :zrem, :zadd, :zremrangebyrank, :zcard, :exists?, :del
+
+  attr_accessor :typed
+  attr_reader :limit
+
+  def elements
+    strings_to_types(zrange(0, -1) || [], typed)
+  end
+  alias to_a elements
+
+  def remove(*elements)
+    zrem(types_to_strings(elements, typed))
+  end
+
+  def prepend(elements)
+    insert(elements, prepending: true)
+  end
+
+  def append(elements)
+    insert(elements)
+  end
+  alias << append
+
+  def limit=(limit)
+    raise "Limit must be greater than 0" if limit && limit <= 0
+
+    @limit = limit
+  end
+
+  private
+    def insert(elements, prepending: false)
+      elements = Array(elements)
+      return if elements.empty?
+
+      elements_with_scores = types_to_strings(elements, typed).map.with_index do |element, index|
+        score = generate_base_score(negative: prepending) + index / 100000
```

Comment:
Could you elaborate what `index / 10000` is for?

Reply:
> Redis sorted sets use a double 64-bit floating point number to represent the score. In all the architectures we support, this is represented as an IEEE 754 floating point number

I was trying to store the maximum unique precision possible in Redis, to store scores (times). By dividing a large integer (seconds) by 100000 we get a float. e.g:

```rb
>> Time.now.to_i
=> 1653898149
>> Time.now.to_f / 100000
=> 16538.98132595902
>> 3.times { p Time.now.to_f / 100000 }
16538.98262287975
16538.982622880398
16538.98262288052
```

Even though we lose some precision, each iteration produces a higher score than the previous one, perfect for our use case of appending.